src/Adapter/Currency/CommandHandler/ScheduleExchangeRatesUpdateHandler.php

```diff
@@ -0,0 +1,247 @@
+
+    /**
+     * @var ShopUrlDataProvider
+     */
+    private $shopUrlDataProvider;
+
+    /**
+     * @var string
+     */
+    private $adminBaseUrl;
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param Configuration $configuration
+     * @param Shop $contextShop
+     * @param ShopUrlDataProvider $shopUrlDataProvider
+     * @param TranslatorInterface $translator
+     * @param bool $isCronJobModuleInstalled
+     * @param string $adminBaseUrl
+     * @param string $dbPrefix
+     */
+    public function __construct(
+        Configuration $configuration,
+        Shop $contextShop,
+        ShopUrlDataProvider $shopUrlDataProvider,
+        TranslatorInterface $translator,
+        $isCronJobModuleInstalled,
+        $adminBaseUrl,
+        $dbPrefix
+    ) {
+        $this->configuration = $configuration;
+        $this->contextShop = $contextShop;
+        $this->isCronJobModuleInstalled = $isCronJobModuleInstalled;
+        $this->shopUrlDataProvider = $shopUrlDataProvider;
+        $this->adminBaseUrl = $adminBaseUrl;
+        $this->dbPrefix = $dbPrefix;
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CurrencyException
+     */
+    public function handle(ScheduleExchangeRatesUpdateCommand $command)
+    {
+        if (!$this->isCronJobModuleInstalled) {
+            throw new ScheduleExchangeRatesUpdateException(
+                'Live exchange rates feature cannot be modified because "cronjob" module is not installed',
+                ScheduleExchangeRatesUpdateException::CRON_TASK_MANAGER_MODULE_NOT_INSTALLED
+            );
+        }
+
+        $this->configuration->restrictUpdatesTo($this->contextShop);
+
+        $cronId = $this->configuration->get('PS_ACTIVE_CRONJOB_EXCHANGE_RATE');
+
+        try {
+            if ($cronId && !$command->isExchangeRateEnabled()) {
+                $this->removeCronJob($cronId);
+
+                return;
+            }
+
+            $cronUrl = $this->getCronUrl();
+
+            if (!$cronId && $command->isExchangeRateEnabled() && false === $this->createCronJob($cronUrl)) {
+                throw new ScheduleExchangeRatesUpdateException(
+                    'Failed to create a cron task for live exchange rate update',
+                    ScheduleExchangeRatesUpdateException::CRON_TASK_CREATION_FAILED
+                );
+            }
+
+            if ($cronId) {
+                $this->resetCronConfiguration($cronId);
+            }
+        } catch (Exception $exception) {
+            throw new CurrencyException(
+                'An unexpected error occurred when trying to update live exchange rates',
+                0,
+                $exception
+            );
+        }
+    }
+
+    /**
+     * @return string
+     *
+     * @throws PrestaShopException
+     */
+    private function getCronUrl()
```

Comment:
But I'm thinking it won't help to remember for someone if anything is added to the LinkCore or new class is being added like `CurrencyRateScriptUrlProvider` because:

If I start refactoring `cron_currency_rates.php` I know it is the file and somehow it should be called from somewhere so I search across whole project to find `cron_currency_rates.php` string
This will lead the way to the handler `ScheduleExchangeRatesUpdateHandler`

If I add To `LinkCore` new method my search method of this file usage will not change - I would do the same :thinking:

Reply:
OK since this is the only place where it is used 👍 I agree
If there was more than 1 place where it would be used, it would be nice to have a method to call 😉